sys/dev_vga.c

```diff
@@ -14,10 +14,54 @@ static int dev_vga_palette_write(vnode_t *v, uio_t *uio) {
   return vga_palette_write((vga_device_t *)v->v_data, uio);
 }
 
+#define RES_CTRL_BUFFER_SIZE 32
+
+static int dev_vga_resolution_write(vnode_t *v, uio_t *uio) {
+  vga_device_t *vga = (vga_device_t *)v->v_data;
+  uio->uio_offset = 0; /* This file does not support offsets. */
+  uint16_t xres, yres;
+  unsigned int bpp;
+  int error = vga_get_resolution(vga, &xres, &yres, &bpp);
+  if (error)
+    return error;
+  char buffer[RES_CTRL_BUFFER_SIZE];
```

Comment:
Because of quite limited kernel stack size, it's a good practice to keep arrays on heap (`vga_device_t` struct?).